src/actions/arcActions.ts

```diff
@@ -568,7 +568,23 @@ export function createProposal(daoAvatarAddress: string, title: string, descript
       const votingMachineParamsHash = await dao.controller.getSchemeParameters(votingMachineInstance.contract.address, dao.avatar.address)
       const votingMachineParams = await votingMachineInstance.contract.parameters(votingMachineParamsHash)
 
-      const submitProposalTransaction: any = await Util.performAction(
+      const descriptionHash = Arc.Utils.SHA3(description);
```

Comment:
Why hash the description?

Reply:
because we dont have the proposalId until the transaction completes so we look up the proposal by descriptionHash onProposalCreateEvent

Reply:
ok